daisy/snapshot_test.go

```diff
@@ -0,0 +1,86 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisy
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func TestSnapshotPopulate(t *testing.T) {
+	w := testWorkflow()
+
+	tests := []struct {
+		desc      string
+		ss        *Snapshot
+		shouldErr bool
+	}{
+		{"default case", &Snapshot{}, false},
+	}
```

Comment:
Looks like there are some additional items in validate that you can include. A few examples:

1. Validating your regex, both in the expected case, and for when the input is malformed
2. Extending the partial URL for the source disk (and not extending)
3. Validating that the link is created as expected

Reply:
added